superset-frontend/src/views/components/MenuRight.tsx

```diff
@@ -144,9 +158,30 @@ const RightMenu = ({
       {menu.label}
     </>
   );
+
+  const handleMenuSelection = (itemChose: React.Key) => {
+    if (itemChose === GlobalMenuDataOptions.DB_CONNECTION) {
+      setShowModal(true);
+      setEngine('');
+    }
+    if (itemChose === GlobalMenuDataOptions.GOOGLE_SHEETS) {
+      setShowModal(true);
+      setEngine('Google Sheets');
+    }
+  };
+
   return (
     <StyledDiv align={align}>
-      <Menu mode="horizontal">
+      <DatabaseModal
+        onHide={() => setShowModal(false)}
+        show={showModal}
+        dbEngine={engine}
+      />
+      <Menu
+        selectable={false}
+        mode="horizontal"
+        onClick={({ key }) => handleMenuSelection(key)}
```

Comment:
another nit, but you can dry this up a bit by just passing the function, and then you can destructure the key in the props above in the function declaration itself.